packages/react/src/ReactFetch.js

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import {
+  enableCache,
+  enableFetchInstrumentation,
+} from 'shared/ReactFeatureFlags';
+
+import ReactCurrentCache from './ReactCurrentCache';
+
+function createFetchCache(): Map<string, Array<any>> {
+  return new Map();
+}
+
+const simpleCacheKey = '["GET",[],null,"follow",null,null,null,null]'; // generateCacheKey(new Request('https://blank'));
+
+function generateCacheKey(request: Request): string {
+  // We pick the fields that goes into the key used to dedupe requests.
+  // We don't include the `cache` field, because we end up using whatever
+  // caching resulted from the first request.
+  // Notably we currently don't consider non-standard (or future) options.
+  // This might not be safe. TODO: warn for non-standard extensions differing.
+  // IF YOU CHANGE THIS UPDATE THE simpleCacheKey ABOVE.
+  return JSON.stringify([
```

Comment:
Since this isn't actually used as a key we could just make it an explicit comparison of two Request objects.

Reply:
Intuitively that seems like it would be faster to me. But if you're not confident either way, seems fine to leave as is and benchmark later.

Reply:
OTOH this is less code